platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts

```diff
@@ -0,0 +1,166 @@
+// class definition of DOMContext.
+class AjaxEngine {
+	private readonly nodes = new Map<number, Node>();
+	private zIndex = 100;
+
+	public allocateZIndex() {
+		this.zIndex += 1;
+		return '' + this.zIndex;
```

Comment:
Why would we force a String here? Seems odd.

Reply:
because the result is assigned to `style.zIndex`, otherwise it would be an error on the caller site